crates/uv-cli/src/lib.rs

```diff
@@ -906,23 +906,26 @@ fn parse_find_links(input: &str) -> Result<Maybe<PipFindLinks>, String> {
     }
 }
 
-/// Parse an `--index` argument into an [`Index`], mapping the empty string to `None`.
-fn parse_index(input: &str) -> Result<Maybe<Index>, String> {
-    if input.is_empty() {
-        Ok(Maybe::None)
-    } else {
-        match Index::from_str(input) {
-            Ok(index) => Ok(Maybe::Some(Index {
+/// Parse an `--index` argument into a [`Vec<Index>`], mapping the empty string to an empty Vec.
+fn parse_indices(input: &str) -> Result<Vec<Maybe<Index>>, String> {
```

Comment:
Might be nice to add some more context here to note that this function exists because clap value_delimiter only lets you split on a single character, and we need to split on all whitespace (notably spaces OR newlines). 

And also that the --index arguments have to have a `Vec<Vec<Maybe<Index>>>` because when clap parses an environment variable with `value_parser` it expects to only parse one value.